src/app/clusters/thread-border-router-management-server/thread-border-router-management-server.cpp

```diff
@@ -259,15 +279,28 @@ CHIP_ERROR ServerInstance::Read(const ConcreteReadAttributePath & aPath, Attribu
         break;
     }
     case Attributes::ActiveDatasetTimestamp::Id: {
-        Optional<uint64_t> activeDatasetTimestamp = ReadActiveDatasetTimestamp();
-        status = activeDatasetTimestamp.HasValue() ? aEncoder.Encode(DataModel::MakeNullable(activeDatasetTimestamp.Value()))
-                                                   : aEncoder.EncodeNull();
+        std::optional<uint64_t> activeDatasetTimestamp = ReadActiveDatasetTimestamp();
+        status = activeDatasetTimestamp.has_value() ? aEncoder.Encode(activeDatasetTimestamp.value()) : aEncoder.EncodeNull();
+        break;
+    }
+    case Attributes::PendingDatasetTimestamp::Id: {
+        std::optional<uint64_t> pendingDatasetTimestamp = ReadPendingDatasetTimestamp();
+        status = pendingDatasetTimestamp.has_value() ? aEncoder.Encode(pendingDatasetTimestamp.value()) : aEncoder.EncodeNull();
         break;
     }
     default:
         break;
     }
-    return status;
+    if (status == CHIP_ERROR_NO_MEMORY)
+    {
+        // If the status returned by Encode function is CHIP_ERROR_NO_MEMORY, map it to IM status ResourceExhausted
+        return CHIP_IM_GLOBAL_STATUS(ResourceExhausted);
```

Comment:
This will break read/subscribe chunking.  Why are we doing this?

Reply:
Returned status now.